homeassistant/components/homekit/type_covers.py

```diff
@@ -96,3 +98,63 @@ def update_state(self, new_state):
                     abs(current_position - self.homekit_target) < 6:
                 self.char_target_position.set_value(current_position)
                 self.homekit_target = None
+
+
+@TYPES.register('WindowCoveringBasic')
+class WindowCoveringBasic(HomeAccessory):
+    """Generate a Window accessory for a cover entity.
+
+    The cover entity must support:
+        - open_cover
+        - close_cover
+        - stop_cover (optional)
+    """
+
+    def __init__(self, *args, config):
+        """Initialize a WindowCovering accessory object."""
+        super().__init__(*args, category=CATEGORY_WINDOW_COVERING)
+        features = self.hass.states.get(self.entity_id) \
+            .attributes.get(ATTR_SUPPORTED_FEATURES)
+        self.supports_stop = features & SUPPORT_STOP
+
+        serv_cover = add_preload_service(self, SERV_WINDOW_COVERING)
+        self.char_current_position = setup_char(
+            CHAR_CURRENT_POSITION, serv_cover, value=0)
+        self.char_target_position = setup_char(
+            CHAR_TARGET_POSITION, serv_cover, value=0,
+            callback=self.move_cover)
+
+    def move_cover(self, value):
+        """Move cover to value if call came from HomeKit."""
+        _LOGGER.debug('%s: Set position to %d', self.entity_id, value)
+
+        service, position = (None, None)
+        if self.supports_stop:
+            if value > 70:
+                service, position = (SERVICE_OPEN_COVER, 100)
+            elif value < 30:
+                service, position = (SERVICE_CLOSE_COVER, 0)
+            elif self.supports_stop:
+                service, position = (SERVICE_STOP_COVER, 50)
+        else:
+            if value >= 50:
+                service, position = (SERVICE_OPEN_COVER, 100)
+            elif value < 50:
+                service, position = (SERVICE_CLOSE_COVER, 0)
+
+        if service is not None:
```

Comment:
This check will always evaluate to true. See above

Reply:
Cool, well i'm happy to omit, but static analysis tools aren't smart enough to make that inference:

![image](https://user-images.githubusercontent.com/1289759/38902559-6852396e-42e4-11e8-9bf7-7a3bfaa07350.png)

Reply:
This should work though or am I getting something wrong here?